File: reactor/r2dbc-mysql/src/main/java/ua/mai/zyme/r2dbcmysql/repository/BalanceRepository.java
package ua.mai.zyme.r2dbcmysql.repository;

import org.springframework.data.r2dbc.repository.Query;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.transaction.annotation.Transactional;
import reactor.core.publisher.Flux;
import ua.mai.zyme.r2dbcmysql.entity.Balance;
import org.springframework.data.r2dbc.repository.R2dbcRepository;
import reactor.core.publisher.Mono;

import java.time.LocalDateTime;

public interface BalanceRepository extends R2dbcRepository<Balance, Integer> {

    // Нет возможности вернуть Mono<Balance>.
    @Query("INSERT INTO balance (member_id, amount, created_date, last_modified_date) VALUES (:memberId, :amount, :createdDate, :lastModifiedDate)")
    Mono<Void> insert(Integer memberId, Long amount, LocalDateTime createdDate, LocalDateTime lastModifiedDate);

    Mono<Balance> findByMemberId(Integer memberId);

    Flux<Balance> findByAmountIsBetween(Long minAmount, Long maxAmount);

}

File: kotlin/kotlin-r2dbc-mysql/kotlin-r2dbc-mysql-be/src/main/java/ua/mai/zyme/r2dbcmysql/repository/BalanceRepository.kt
package ua.mai.zyme.r2dbcmysql.repository

import org.springframework.data.r2dbc.repository.Query
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import ua.mai.zyme.r2dbcmysql.entity.Balance
import org.springframework.data.r2dbc.repository.R2dbcRepository
import java.time.LocalDateTime

interface BalanceRepository : R2dbcRepository<Balance, Int> {

    // Нет возможности вернуть Mono<Balance>.
    @Query("INSERT INTO balance (member_id, amount, created_date, last_modified_date) VALUES (:memberId, :amount, :createdDate, :lastModifiedDate)")
    fun insert(memberId: Int, amount: Long, createdDate: LocalDateTime, lastModifiedDate: LocalDateTime): Mono<Void>

    fun findByMemberId(memberId: Int): Mono<Balance>

    fun findByAmountIsBetween(minAmount: Long, maxAmount: Long): Flux<Balance>

}